### PR TITLE
SQL-1557: Update SQLDriverConnectW to just return SUCCESS rather than…

### PR DESCRIPTION
**Pull request by pmeredit**
… SUCCESS_WITH_INFO when the out_connection_string is NULL or buffer_len is 0

This is passing locally. The resultset tests show that we no longer return SUCCESS_WITH_INFO when a null buffer/0 buffer length are passed.

### CODE REVIEW

**Review by terakilobyte — COMMENTED** [review]
This looks like the ticket. I'm going to checkout the pr in ubuntu and test it with `iusql`

**Review by terakilobyte — APPROVED** [review]
LGTM. Confirmed this works locally.

**Comment by pmeredit** [comment]
evergreen merge

File: odbc/src/api/functions.rs [hunk 1]
```diff
@@ -930,6 +930,7 @@ pub unsafe extern "C" fn SQLDriverConnectW(
     string_length_2: *mut SmallInt,
     driver_completion: DriverConnectOption,
 ) -> SqlReturn {
+    dbg!("!!!!");
```
Comment (nbagnard) [inline odbc/src/api/functions.rs]: Left over debug.

File: integration_test/tests/common/mod.rs [hunk 2]
```diff
@@ -150,7 +150,18 @@ pub fn connect_with_conn_string(env_handle: HEnv, in_connection_string: String)
             str_len_ptr,
             DriverConnectOption::NoPrompt,
         ) {
-            SqlReturn::SUCCESS | SqlReturn::SUCCESS_WITH_INFO => (),
+            // Originally, this would return SUCCESS_WITH_INFO since we pass null_mut() as
```
Comment (nbagnard) [inline integration_test/tests/common/mod.rs:153]: I don't think that there is much value in keep some kind of historical comment on a test.
If you want to track the changes, I'd suggest to tag the comment in the feature change itself (in function.rs) with the JIRA ticket which introduced the change.
This help track changes overtime and now which version it was introduced into.

**Comment by pmeredit** [comment]
evergreen push

**Comment by pmeredit** [comment]
evergreen merge

**Comment by pmeredit** [comment]
> evergreen push

Heh, where was my mind?

**Pull request merged**